runners/apex/src/main/java/org/apache/beam/runners/apex/translation/ParDoTranslator.java

```diff
@@ -77,6 +80,13 @@ public void translate(ParDo.MultiOutput<InputT, OutputT> transform, TranslationC
     PCollection<InputT> input = context.getInput();
     List<PCollectionView<?>> sideInputs = transform.getSideInputs();
 
+    DoFnSchemaInformation doFnSchemaInformation;
+    try {
+      doFnSchemaInformation = ParDoTranslation.getSchemaInformation(context.getCurrentTransform());
+    } catch (IOException e) {
```

Comment:
This exception handling is duplicated all over. Should it be moved into `getSchemaInformation`? When does the IOException get raised? Is there any value in forcing callers to handle it? If you wouldn't make `getSchemaInformation` return a `oneof` then it shouldn't throw a checked exception.

Reply:
Done